### PR TITLE
🎉 New Source: Okta

### LINKED ISSUE #2414: Source: Okta
## Tell us about the new integration you’d like to have
* Which source and which destination?
* Do you need a specific version of the underlying data source e.g: you specifically need support for an older version of the API or DB? 

## Describe the context around this new integration
* Which team in your company wants this integration, what for? This helps us understand the use case.
* How often do you want to run syncs?
* If this is an API source connector, which entities/endpoints do you need supported?

## Describe the alternative you are considering or using
What are you considering doing if you don’t have this integration through Airbyte?

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368094671073) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by grebessi**
## What
Added new source to Okta.
This PR closes #2414

## How
Added support for the following schemas:

* Users
* Groups
* System Log

## Pre-merge Checklist
* [ ] _Run integration tests_
* [ ] _Publish Docker images_

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200368060087111/1200375175244594) by [Unito](https://www.unito.io)

### CODE REVIEW

File: airbyte-integrations/connectors/source-okta/build.gradle [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    id 'airbyte-source-acceptance-test'
+}
+
+airbytePython {
+    moduleDirectory 'source_okta'
+}
+
+dependencies {
+    implementation files(project(':airbyte-integrations:bases:source-acceptance-test').airbyteDocker.outputs)
+    implementation files(project(':airbyte-integrations:bases:base-python').airbyteDocker.outputs)
```
Comment (keu) [inline airbyte-integrations/connectors/source-okta/build.gradle:13]: should not be necessary anymore

File: airbyte-integrations/connectors/source-okta/integration_tests/catalog.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    {
+      "stream": {
+        "name": "users",
+        "json_schema": {}
+      },
+      "sync_mode": "incremental",
+      "destination_sync_mode": "overwrite"
+    }
+  ]
+}
```
Comment (keu) [inline airbyte-integrations/connectors/source-okta/integration_tests/catalog.json:28]: newline

File: airbyte-integrations/connectors/source-okta/sample_files/catalog.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    {
+      "stream": {
+        "name": "users",
+        "json_schema": {}
+      },
+      "sync_mode": "incremental",
+      "destination_sync_mode": "overwrite"
+    }
+  ]
+}
```
Comment (keu) [inline airbyte-integrations/connectors/source-okta/sample_files/catalog.json:28]: nl

File: airbyte-integrations/connectors/source-okta/setup.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+from setuptools import find_packages, setup
+
+MAIN_REQUIREMENTS = [
+    "airbyte-cdk",
```
Comment (keu) [inline airbyte-integrations/connectors/source-okta/setup.py:27]: bind version ~=0.1

File: airbyte-integrations/connectors/source-okta/source_okta/source.py [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+        return None
+
+    def request_params(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+        return {
+            'limit': 200,
```
Comment (keu) [inline airbyte-integrations/connectors/source-okta/source_okta/source.py:68]: lets move to class attribute

File: airbyte-integrations/connectors/source-okta/source_okta/source.py [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+            **(next_page_token or {}),
+        }
+
+    def parse_response(self,
+        response: requests.Response,
+        **kwargs,
+    ) -> Iterable[Mapping]:
+        for row in response.json():
+            yield row
```
Comment (keu) [inline airbyte-integrations/connectors/source-okta/source_okta/source.py:77]:
```suggestion
        yield from response.json()
```

**Review by keu — CHANGES_REQUESTED** [review]
minor changes

**Review by keu — APPROVED** [review]

File: airbyte-integrations/connectors/source-okta/sample_files/configure_catalog.json [hunk 7]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-okta/sample_files/configure_catalog.json:1]: this seems not to belong to this connector

File: airbyte-integrations/connectors/source-okta/source_okta/source.py [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+
+
+class OktaStream(HttpStream, ABC):
+    def __init__(self, url_base: str, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # Inject custom url base to the stream
+        self._url_base = url_base
+    
+    @property
+    def page_size(self) -> int:
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-okta/source_okta/source.py:44]: why is this not a hardcoded property e.g: `page_size = 200`?

File: airbyte-integrations/connectors/source-okta/sample_files/catalog.json [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+      },
+      "sync_mode": "incremental",
+      "destination_sync_mode": "overwrite"
+    },
+    {
+      "stream": {
+        "name": "logs",
+        "json_schema": {}
+      },
+      "sync_mode": "incremental",
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-okta/sample_files/catalog.json:16]: this catalog used for testing should contain the cursor field and primary key for each stream

**Review by sherifnada — APPROVED** [review]
Very small comments. Please bump and publish once you've addressed them. 

Also, please change the PR name to `🎉 New Source: Okta`

**Review by sherifnada — CHANGES_REQUESTED** [review]
Actually just realized a few things:
1. Which API token are you using? I thought Okta was an Oauth based source
2. Could you add the config to Github secrets, `ci_credentials.sh`, `test-command` and `publish-command.yml`? 
3. we need to add docs for this connector in the `docs/` directory. Could you pattern match off the existing connectors to do this? Also, we need to update `SUMMARY.md` to point to these docs.

**Comment by grebessi** [comment]
> Actually just realized a few things:
> 
> 1. Which API token are you using? I thought Okta was an Oauth based source
> 2. Could you add the config to Github secrets, `ci_credentials.sh`, `test-command` and `publish-command.yml`?
> 3. we need to add docs for this connector in the `docs/` directory. Could you pattern match off the existing connectors to do this? Also, we need to update `SUMMARY.md` to point to these docs.

1. According to [the docs](https://developer.okta.com/docs/reference/api/authn/#primary-authentication), we can authenticate in 5 ways:
- Public Application
- [Trusted Application](https://developer.okta.com/docs/reference/api/authn/#primary-authentication-with-trusted-application)
- Activation Token
- Device Fingerprinting
- Password Expiration Warning
In this case, I did use `Trusted Application`. I used the token provided by `LastPass` [(Create the token docs)](https://developer.okta.com/docs/guides/create-an-api-token/create-the-token/)

2 and 3, I will start working on adding it.

**Review by po3na4skld — APPROVED** [review]

File: docs/integrations/README.md [hunk 10]
```diff
@@ -45,6 +45,7 @@ Airbyte uses a grading system for connectors to help users understand what to ex
 |[Mixpanel](./sources/mixpanel.md)| Beta |
 |[Mongo DB](./sources/mongodb.md)| Alpha |
 |[MySQL](./sources/mysql.md)| Certified |
+|[Okta](./sources/oracle.md)| Beta |
```
Comment (sherifnada) [inline docs/integrations/README.md:48]:
```suggestion
|[Okta](./sources/okta.md)| Beta |
```

**Review by sherifnada — APPROVED** [review]
LGTM -- please publish and merge

**Pull request merged**